### ISSUE TITLE
Wpf: occasional "'Hyperlink' is on a Visual a Visual3D" exception in TreeGridView.ReloadData() 

### ISSUE THREAD

**Issue by msasso69** [picoe/Eto#814]
Occasionally I get the "'SystemWindows.Document.Hyperlink' is not a Visual or a Visual3D" exception while invoking the method RelodData in the TreeGridView.
I have not been able to sistematically reproduce it in the simple test code, but I managed to fix it as follows:

	public static bool HasFocus(this sw.DependencyObject control, sw.DependencyObject focusScope, bool checkChildren = true)
	{
		var current = swi.FocusManager.GetFocusedElement(focusScope) as sw.DependencyObject;
		if (!checkChildren)
			return current == control;

		while (current != null)
		{
			if (current == control)
				return true;
	ADD THIS ->	if (current is System.Windows.Documents.Hyperlink)
	ADD THIS ->		return false;
			current = swm.VisualTreeHelper.GetParent(current);
		}
		return false;
	}

Basically simply add the above two lines in the method HasFocus of the class WpfExtensions in Source\Eto.Wpf\WpfExtensions.cs and the exception will never be fired :)

**Comment by cwensley** [comment]
Thanks for the report (and the proposed fix).  I think I have a more generic way to handle this (not just with hyperlink), so I may not be able to use your PR directly but I'll get this fixed up.

Cheers!
Curtis.

**Comment by cwensley** [comment]
Thanks again for reporting the issue!  This is now fixed and should handle any content elements if they have focus.  I managed to replicate the crash with the unit test [here](https://github.com/picoe/Eto/commit/0b2f41e877f634be75d2e8a55eab47f0e2605682#diff-ca8d8baa4f081a4aac21bf921d8f1671R30).

**Comment by msasso69** [comment]
Thanks Curtis, in the meantime I made my fix more generic as you suggested and posted it to you as a comment in the PR #821 (not sure you noticed it though), your fix looks similar and I'm now using it, I will let you know if it works also in my code. Thanks a lot again anyway for your support and your amazing job! :)
